frontend/src/services/FakeDataSchemaService.ts

```diff
@@ -0,0 +1,123 @@
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import { EdgeType } from '../shared/schema/EdgeType';
+import { EntityType } from '../shared/schema/EntityType';
+import { EntityTypeAttribute } from '../shared/schema/EntityTypeAttribute';
+import { NodeType } from '../shared/schema/NodeType';
+import { CancellationToken } from '../utils/CancellationToken';
+import delay from '../utils/delay';
+import nouns from '../utils/nouns';
+import withCancellation from '../utils/withCancellation';
+import getRandomIndex from '../utils/getRandomIndex';
+import SchemaService from './SchemaService';
+
+// TODO: What else is possible?
+export const possibleAttributeTypes: string[] = [
+  'String',
+  'Long',
+  'StringArray',
+];
+
+@injectable()
+export class FakeDataSchemaService extends SchemaService {
+  private readonly usedNounIndices: Set<number> = new Set<number>();
+
+  private edgeTypesPromise: Promise<EdgeType[]> | null = null;
+
+  private nodeTypesPromise: Promise<NodeType[]> | null = null;
+
+  private getRandomNoun(): string {
+    let nounIndex = getRandomIndex(nouns.length);
+
+    while (this.usedNounIndices.has(nounIndex)) {
+      nounIndex = getRandomIndex(nouns.length);
+    }
+
+    this.usedNounIndices.add(nounIndex);
+    return nouns[nounIndex];
+  }
+
+  private generateRandomEntityTypeAttribute(): EntityTypeAttribute {
+    const name = this.getRandomNoun();
+
+    // Build the string array that contains the types the attribute may contain.
+    // The possible types are stored in a constant.
+    const types: string[] = [];
+
+    // Get a random number in the range [1..possibleAttributeTypes.length[ that determines
+    // the amount of types that the array will contain.
+    const numberOfTypes = getRandomIndex(possibleAttributeTypes.length - 1) + 1;
+
+    // A copy of the possibleAttributeTypes, as we modify this list now.
+    const nonAllocatedAttributeTypes = [...possibleAttributeTypes];
+
+    for (let i = 0; i < numberOfTypes; i += 1) {
+      // First get an index into the collection of not yet allocated attribute types.
+      const idx = getRandomIndex(nonAllocatedAttributeTypes.length);
+
+      // Add the attribute type to the collection.
+      types.push(nonAllocatedAttributeTypes[idx]);
+
+      // Remove the attribute type from the collection of non allocated attribute types,
+      // as we just allocated it.
+      nonAllocatedAttributeTypes.splice(idx, 1);
+    }
+
+    const mandatory = getRandomIndex(1) === 1;
+
+    return { name, types: <[string]>types, mandatory };
+  }
+
+  private generateRandomEntityType(): EntityType {
+    const name = this.getRandomNoun();
+    const numberOfAttributes = getRandomIndex(20);
```

Comment:
maybe take the "20" as an input-parameter "numberOfAttributesRange" or something